erigon-lib/recsplit/deltaseq/delta_seq.go

```diff
@@ -0,0 +1,170 @@
+package deltaseq
+
+import (
+	"encoding/binary"
+	"sort"
+
+	"github.com/erigontech/erigon-lib/kv/stream"
+)
+
+type DeltaSeq struct {
+	baseNum   uint64
+	rawDeltas []byte
+	pos       int
+}
+
+func NewDeltaSeq(baseNum uint64, count uint64) *DeltaSeq {
+	return &DeltaSeq{
+		baseNum:   baseNum,
+		rawDeltas: make([]byte, count*4),
+		pos:       0,
+	}
+}
+
+func ReadDeltaSeq(baseNum uint64, rawDeltas []byte) *DeltaSeq {
+	return &DeltaSeq{
+		baseNum:   baseNum,
+		rawDeltas: rawDeltas,
+		pos:       len(rawDeltas) / 4,
+	}
+}
+
+func (s *DeltaSeq) Get(i uint64) uint64 {
```

Comment:
1. `delta encoding` has a bit different meaning: https://en.wikipedia.org/wiki/Delta_encoding 
2. `Uint32(s.rawDeltas[i*4 : (i+1)*4])` it means that: max allowed data - is `_max - _min < MaxU32`. It's interesting condition. It's good for small sequences. Did you check `MaxU16` - is worse?

Reply:
1 - out of ideas, let's rename it to `simpleseq.SimpleSequence`

Reply:
2 - the range of txNums inside 1 .ef is bound by endStep - startStep, which is hardcoded to a max of 64 steps currently, that means the difference is at most 100M txs, so it is safe to use uint32 as long as the hardcoded constants don't change inside erigon.

but it is not safe to use uint16.